src/main/java/org/terasology/gookeeper/system/EconomySystem.java

```diff
@@ -75,32 +93,120 @@
     @In
     private NUIManager nuiManager;
 
+    @In
+    private BlockCommands blockCommands;
+
+    @In
+    private ItemCommands itemCommands;
+
     private static final Logger logger = LoggerFactory.getLogger(EconomySystem.class);
     private Random random = new FastRandom();
     private static final float baseEntranceFee = 100f;
     private static final float baseVisitFee = 10f;
-    private static boolean setHud = false;
 
     @Override
     public void initialise() {
     }
 
+    @ReceiveEvent
+    public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, InventoryComponent inventory) {
+        if (player.hasComponent(EconomyComponent.class)) {
+            nuiManager.getHUD().addHUDElement("PlayerHud");
+        }
+    }
+
     @Override
     public void shutdown() {
     }
 
     @Override
     public void update(float delta) {
-        // Have to add the wallet hud this way since adding in within initialise() results in the economy component to be null
-        if (!setHud) {
-            for (EntityRef wallet : entityManager.getEntitiesWith(EconomyComponent.class)) {
-                EconomyComponent economyComponent = wallet.getComponent(EconomyComponent.class);
-                if (economyComponent != null) {
-                    nuiManager.getHUD().addHUDElement("PlayerHud");
-                    setHud = true;
+    }
+
+    /**
+     * This command can be used to purchase utility blocks (such as pen blocks, visit blocks, etc.)
+     *
+     * @param client
+     * @param itemPrefabName
+     * @param amount
+     * @param shapeUriParam
+     * @return
```

Comment:
Even though it's a console command I'd still like to see these filled out.

If you feel that they don't need to be there then remove the tags and just have the description.

Either way, blank `@param` and `@return` is not good